modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -221,125 +220,116 @@ export const insertHtmlAtCaret = (editor: Editor, value: string, details: Insert
   }, editor.schema);
   const bookmarkHtml = '<span id="mce_marker" data-mce-type="bookmark">&#xFEFF;</span>';
 
-  // Run beforeSetContent handlers on the HTML to be inserted
-  preProcessSetContent(editor, { content: value, format: 'html', set: false, selection: true, paste: details.paste }).each((args) => {
```

Comment:
There's a few cleanup steps that are run before here, should they also be included in `InsertContent.ts`? (mainly just looking at the whitespace cleanup)